ui/src/app/pages/signed-in/inactivity-monitor.tsx

```diff
@@ -192,33 +193,24 @@ export const InactivityMonitor = () => {
 
     return () => {
       clearInterval(inactivityInterval);
-      [getUserActivityTimer(), logoutTimer, inactivityModalTimer].forEach((t) =>
-        clearTimeout(t)
-      );
+      [getUserActivityTimer(), logoutTimer].forEach((t) => clearTimeout(t));
     };
   }, []);
 
-  const secondsBeforeDisplayingModal =
-    environment.inactivityTimeoutSeconds -
-    environment.inactivityWarningBeforeSeconds;
+  const onCloseInactivityModal = () => {
+    resetSignOutForInactivityTime();
+    // todo: clicking to close the modal will fire a click event, which will reset the timer,
+    // however, we should still reset the timers here so we don't rely on that implicit behavior
+  };
 
   return (
-    <React.Fragment>
-      {showModal && (
-        <TextModal
-          closeFunction={() => setShowModal(false)}
-          title='Your session is about to expire'
-          body={
-            `You have been idle for over ${secondsToText(
-              secondsBeforeDisplayingModal
-            )}. ` +
-            'You can choose to extend your session by clicking the button below. You will be automatically logged ' +
-            `out if there is no action in the next ${secondsToText(
-              environment.inactivityWarningBeforeSeconds
-            )}.`
-          }
-        />
-      )}
-    </React.Fragment>
+    <InactivityModal
+      closeFunction={onCloseInactivityModal}
+      currentTimeMs={currentTimeMs}
```

Comment:
Just a minor question, why do you want to pass currentTime and inactivityWarningBeforeMs to the InactivityModal while you can get them directly in the modal?

Reply:
Great question.

1. The main reason was to maximize testability of `InactivityModal`. Since all of the props are explicitly passed in, the tests were easy to write. There are no tricky dependencies that need to be mocked, like time or global config.
2. I wanted to use `currentTime` in the other inactivity-monitor logic. The inactivity monitor currently has several timers that interact in complex ways and I suspect the entire component can be reduced to using a single timer. However, I didn't have enough time to complete this in a day.
3. I want to isolate the complex parts of the code (time logic, reading global config variables, etc.) in one component.

I'm open to moving either. I feel more strongly about `currentTime` than  `inactivityWarningBeforeMs` since I think the timer logic is responsible for the majority of the complexity of this component and want to keep that isolated.

Reply:
I merged the PR, but let me know if you'd like me to follow up with any changed.

Reply:
Thanks Peter for the thoughts, I agree with the last statement, currentTime could indeed stay inside activity-monitor, while inactivityWarningBeforeMs seems better suited in the modal.

Reply:
Addressed in https://github.com/all-of-us/workbench/pull/8148.

Reply:
I agree that this is better. All uses of `inactivityWarningBeforeSeconds` are now in the same component.